### PR TITLE
Implement platform pay payment method capturing for web

### PR DESCRIPTION
**Pull request by cornwe19**
* Addresses #1122 
* Includes changes for detecting platform pay availability on web as well. (see #1478)

### CODE REVIEW

**Comment by cornwe19** [comment]
@jonasbark @jamesblasco @remonh87 this supersedes #1478

**Comment by cornwe19** [comment]
@jonasbark @jamesblasco @remonh87 looking for feedback on this PR. It addresses a sizeable feature gap that exists on web.

File: packages/stripe_platform_interface/lib/src/stripe_platform_interface.dart [hunk 1]
```diff
@@ -99,6 +99,7 @@ abstract class StripePlatform extends PlatformInterface {
   /// Check if either google pay or apple pay  is supported on device.
   Future<bool> isPlatformPaySupported({
     IsGooglePaySupportedParams? params,
+    PlatformPayWebPaymentRequestCreateOptions? paymentRequestOptions,
```
Comment (remonh87) [inline packages/stripe_platform_interface/lib/src/stripe_platform_interface.dart:102]: I do not like the fact we need to inject a web specific method into the platform interface this should be independent.  Is this really required for just checking if platformpay is supported?

**Review by remonh87 — COMMENTED** [review]
Many thanks for this PR. I did review it and in general it looks good. However I am not convinced about adding the web paymentrequest options to isPlatformPaysupported interface. @jonasbark  what are your thoughts about it?

File: packages/stripe_platform_interface/lib/src/stripe_platform_interface.dart (hunk 1 above)
Comment (cornwe19) [reply packages/stripe_platform_interface/lib/src/stripe_platform_interface.dart:102]: @remonh87 thoughts on my explanation above? Do you have suggestions for a different approach here given the constraints?

Comment (remonh87) [reply packages/stripe_platform_interface/lib/src/stripe_platform_interface.dart:102]: no looks to me. I noticed some conflicts in this branch but we are good to go. @jonasbark  do you want to have a look at it as well?

**Review by remonh87 — APPROVED** [review]

File: packages/stripe_platform_interface/lib/src/stripe_platform_interface.dart (hunk 1 above)
Comment (cornwe19) [reply packages/stripe_platform_interface/lib/src/stripe_platform_interface.dart:102]: Excellent. Thanks for your input. I'll get the conflicts cleared up shortly

File: packages/stripe_web/lib/src/widgets/platform_pay_button.dart [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+const kPlatformPayButtonDefaultHeight = 40.0;
+
+class WebPlatformPayButton extends StatefulWidget {
+  final PlatformPayWebPaymentRequestCreateOptions paymentRequestCreateOptions;
+  final PlatformButtonType? type;
+  final PlatformButtonStyle? style;
+  final BoxConstraints? constraints;
+  final ui.VoidCallback onPressed;
+
+  const WebPlatformPayButton(
```
[the remark this answers is not in the document]
Comment (cornwe19) [reply packages/stripe_web/lib/src/widgets/platform_pay_button.dart:13]: Updated

**Comment by cornwe19** [comment]
@remonh87 I've resolved conflicts and reformatted code based on your suggestions. Let me know if there are any other outstanding items on this one. Excited to get this merged through

c.c. @jonasbark

**Comment by remonh87** [comment]
Thanks for the work. I am running the CI now and give Jonas till tomorrow to have a look at it else I will merge it.

**Comment by cornwe19** [comment]
@remonh87 Thanks!

I'm seeing some failures like `IntegrationError: Please call Stripe() with your publishable key. You used an empty string.` in stripe_js tests. That doesn't immediately feel like something I could have broken with this PR, but please let me know if there's some mitigation required here.

**Comment by cornwe19** [comment]
@remonh87 any remaining items blocking this pr?

**Comment by cornwe19** [comment]
@remonh87 Just synced this PR with latest main. I suspect checks will need to be rerun and will probably fail in similar ways to how they did on Jan, 24th. Please let me know if there's any action needed here. Hoping to get this PR merged soon.

Thanks!

**Comment by jaxnz** [comment]
Excited to see the PR to be merged

**Comment by cornwe19** [comment]
@remonh87 FYI I just realized the way I had implemented payment request cancellation wasn't aligned with how iOS and Android implement it. Added a commit to update that (it now uses a `StripeException` with `LocalizedErrorMessage` having failure code `FailureCode.Canceled`)

Looking forward to getting this merged.

**Comment by haomingzhang0101** [comment]
Thanks for the great work! I am wondering when this feature will be released.

**Comment by manandhar-aarya** [comment]
Hello @cornwe19,
One question:  what is the way to confirm the payment with the client secret in this flow of PlatformPayWebPaymentItem? It says confirmPlatformPayPaymentIntent is not supported on the web and I couldnt find a function that allows to confirm the payment. 
Thanks!

**Pull request merged**